libraries/base/Data/Semigroup.hs

```diff
@@ -18,20 +18,33 @@
 -- Maintainer  :  libraries@haskell.org
 -- Stability   :  provisional
 -- Portability :  portable
+-- 
+-- A type @a@ is a 'Semigroup' if it provides an associative function ('<>')
+-- that lets you combine any two values of type @a@ into one. 
 --
--- In mathematics, a semigroup is an algebraic structure consisting of a
--- set together with an associative binary operation. A semigroup
--- generalizes a monoid in that there might not exist an identity
--- element. It also (originally) generalized a group (a monoid with all
--- inverses) to a type where every element did not have to have an inverse,
--- thus the name semigroup.
+-- Intuitively, this gives rise to an abstraction that allows for the
+-- combination of multiple values into a single one. The instance of 
+-- the 'Semigroup' determines the way in which the values are to be 
+-- combined.
```

Comment:
This paragraph doesn't seem to add anything that the previous paragraph didn't already cover.

Reply:
maybe you are right - i was so stuck on trying to make it "approachable" but maybe this is just redundant. Unless anyone has any ideas on talking about the motivation / usecases...